push_as_submodule.sh

```diff
@@ -0,0 +1,149 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+#----------------------------------
+# parameters to configure by the user
+
+declare -a all_sclorg_images=(\
+"s2i-base"
+"mysql"
+"mariadb"
+"postgresql"
+"redis"
+"s2i-php"
+"s2i-python"
+"s2i-perl"
+"s2i-ruby"
+"varnish"
+"nginx"
+"s2i-nodejs"
+"httpd"
+)
+
+branch=master
+remote=origin
+commit_msg="Update common submodule to current latest $remote/$branch commit"
+
+#----------------------------------
+
+function exit_on_err() {
+  local ret=$1
+  local reason=${2:-"unknown"}
+  if [[ $ret -ne 0 ]]; then
+    echo "ERR: Exiting for reason: $reason" 2>&1 && exit 42
+  fi
+}
+
+function clone_repo() {
+  local image=$1
+  [[ -z $image ]] && exit_on_err 1 "image name for cloning repository is missing"
+  local container="${image}-container"
+  local url="git@github.com:sclorg/${container}.git"
+  git clone "$url" "$image"
+  exit_on_err $? "git clone failed"
+  cd "$image" || return 1
+  git submodule update --init
+  exit_on_err $? "submodule init failed"
+  cd ..
+}
+
+function update_submodule() {
+  local sha=$1
+  [[ -z "$sha" ]] && exit_on_err 1 "commit SHA is missing for updating submodule"
+  cd common/ || return 1
+  git fetch -a
+  git checkout "$sha"
+  exit_on_err $? "checkout common on $sha failed"
+  cd ..
+  git add common/
+}
+
+function commit_change() {
+  local commit_msg=$1
+  [[ -z "$commit_msg" ]] && exit_on_err 1 "commit message is missing"
+  git commit -m "$commit_msg"
+}
+
+function show_diff() {
+  local image=$1
+  local answer
+  git diff $branch $remote/$branch
+  exit_on_err $? "diff failed"
+  echo "Are you sure, you want to push the above changes to $remote/$branch of"
+  echo "git@github.com:sclorg/${image}-container.git?"
+  echo "Y/n"
+  read -r answer
+  while [[ $answer != Y && $answer != n ]]; do
+    read -r answer
+  done
+  [[ "$answer" == "n" ]] && return 1
+  return 0
+}
+
+function push_change() {
+  local image=$1
+  show_diff "$image"
+  [[ $? -eq 1 ]] && echo "WARN: $image will not be pushed." && return 1
+  git push
+  exit_on_err $? "push failed"
+}
+
+function get_remote_branch_hash() {
+  git fetch "$remote"
+  git rev-parse "$remote"/"$branch"
+  exit_on_err $?
+}
+
+function cleanup() {
+  echo "Updated container images are: $updated_containers"
```

Comment:
Would it be possible to remember all pull requests and show them at the end?
Only RFE.

Reply:
This will not create PRs. This will push changes directly to master. Is this a problem?